src/cmt_decode_prometheus.c

```diff
@@ -454,7 +457,7 @@ static int add_metric_histogram(struct cmt_decode_prometheus_context *context)
      * - sum
      * - count */
     bucket_count = mk_list_size(&context->metric.samples) - 3;
-    timestamp = 0;
+    timestamp = context->opts.override_timestamp;
```

Comment:
Needn't add this for add_metric_gauge too?

Reply:
No need, I only needed a special case for histogram/summary, but the others share a common code path (see the tests which contains a counter example)